src/index.js

```diff
@@ -289,11 +291,12 @@ export default function undoable (reducer, rawConfig = {}) {
           }
         }
 
-        if (history.present !== res) {
+        // Don't add entry to history if it was just created
+        if ((justCreatedHistory && history.present === res) || action.type === '@@redux-undo/INIT') {
```

Comment:
gaearon has stated in https://github.com/reactjs/redux/issues/186 that one day they may change the name of `@@redux-undo/INIT` to a randomized action name, so if we check it in any way it could be broken by an update in Redux.

If we use a flag that is set to `true` after the first run of the reducer, then a hot-reload which calls `@@redux/INIT` will have an incorrect flag and will call an extra `insert`.

I'm still struggling with this one :ghost: